Math/NumberTheory/GaussianIntegers.hs

```diff
@@ -192,7 +192,7 @@ a .^ e
 factorise :: GaussianInteger -> [(GaussianInteger, Int)]
 factorise g = helper (Factorisation.factorise $ norm g) g
     where
-    helper [] g' = if g' == 1 then [] else [(g', 1)] -- include the unit, if it isn't 1
+    helper [] g' = if abs g' == 1 then [] else [(g', 1)] -- include the unit, if it isn't 1
```

Comment:
When the first argument is an empty list, `abs g'` is guaranteed to be `1`, so we can just return `[]`. 
In fact, now `helper` do not need to carry the second argument at all.

Reply:
I've reduced the base case to

    helper [] _ = []

but I'm less certain about how to rework the other case to remove the need for `g'`, assuming that's what you mean.

Reply:
My fault, it is non-trivial to remove the need for `g'`. Definitely out-of-scope for this PR.